spring-rest2ts-spring/src/main/java/com/blueveery/springrest2ts/implgens/BaseImplementationGenerator.java

```diff
@@ -89,11 +92,76 @@ protected String getPathFromRequestMapping(RequestMapping requestMapping) {
         return "";
     }
 
-    protected void replaceInStringBuilder(StringBuilder pathStringBuilder, String targetToReplace, String replacement) {
-        int start = pathStringBuilder.lastIndexOf(targetToReplace);
-        validatePathVariableParameter(pathStringBuilder, targetToReplace, start);
-        int end = start + targetToReplace.length();
-        pathStringBuilder.replace(start, end, replacement);
+    protected void replaceInStringBuilder(StringBuilder pathStringBuilder, String pathVariableName, TSParameter tsParameter) {
+        final String captureTheRest = "{*" + pathVariableName + "}";
+        final String captureVariable = "{" + pathVariableName + "}";
+
+        stripPatternParserFeatures(pathStringBuilder);
+        
+        int startCaptureVariable = pathStringBuilder.lastIndexOf(captureVariable);
+        int startCaptureTheRest = pathStringBuilder.lastIndexOf(captureTheRest);
+        
+        if (startCaptureTheRest != -1) {
+            if (tsParameter.getType() instanceof TSArray) {
+                pathStringBuilder.replace(startCaptureTheRest,
+                    startCaptureTheRest + captureTheRest.length(),
+                    "' + " + pathVariableName +
+                    ".flatMap(x=>encodeURIComponent(x)).join('/') + '");
+            } else {
+                pathStringBuilder.replace(startCaptureTheRest,
+                    startCaptureTheRest + captureTheRest.length(),
+                    "' + encodeURIComponent(" + pathVariableName + ") + '");
+            }
+        } else if (startCaptureVariable != -1) {
+            pathStringBuilder.replace(startCaptureVariable,
+                startCaptureVariable + captureVariable.length(),
+                "' + encodeURIComponent(" + pathVariableName + ") + '");
+        } else {
+            validatePathVariableParameter(pathStringBuilder, pathVariableName, -1);  
+        }
+
+    }
+
+    protected static void stripPatternParserFeatures(StringBuilder pathStringBuilder)
+    {
+      stripPatternParserFeatures(pathStringBuilder, 0);
+    }
+
+    protected static void stripPatternParserFeatures(StringBuilder pathStringBuilder, int fromIndex)
```

Comment:
I didn't simply call this strip RegexPattern because again I am not certain what Spring wanted to do with this.  It is possible perhaps that they may add some sort of SpEL expression feature here in the future, not sure.